### PR TITLE
Add opentelemetry 0.23 feature, crates and tests

### PR DESCRIPTION
**Pull request by cottinisimone**

### CODE REVIEW

**Review by cpiemontese — DISMISSED** [review]
:cry: really sad that we have to do this thing

File: Cargo.toml [hunk 1]
```diff
@@ -11,16 +11,17 @@ version = "0.16.3"
 rust-version = "1.72"
 
 [features]
-default = ["tracing_opentelemetry"]
+default = []
```
Comment (cottinisimone) [inline Cargo.toml:14]: I removed the default feature. I have the feeling that the user must be aware of the otel version he's using rather than have it by magic. 

Wdyt?

If you do agree i would remove `tracing_opentelemetry` feature or rename it to `latest_tracing_opentelemetry`. Revert this change otherwise

**Review by MaeIsBad — COMMENTED** [review]
Update changelog and bump?

**Review by MaeIsBad — APPROVED** [review]
